### PR TITLE
Ref topup. Fix sendmax

### PR DESCRIPTION
**Pull request by cmgustavo**

### CODE REVIEW

**Comment by cmgustavo** [comment]
Updated. Ready for review

File: src/js/services/txFormatService.js [hunk 1]
```diff
@@ -23,6 +23,26 @@ angular.module('copayApp.services').factory('txFormatService', function($filter,
     return root.formatAmount(satoshis) + ' ' + config.unitName;
   };
 
+  root.formatToCode = function(satoshis, code, cb) {
```
Comment (matiu) [inline src/js/services/txFormatService.js]: maybe a better name would be "toFiat"

File: src/js/controllers/topup.js [hunk 2]
```diff
@@ -50,29 +60,146 @@ angular.module('copayApp.controllers').controller('topUpController', function($s
...
+        return cb({
+          title: gettextCatalog.getString('Could not create transaction'),
+          message: bwcError.msg(err)
+        });
+      }
+      return cb(null, ctxp);
+    });
+  };
+
+  var parseAmount = function(wallet, cb) {
```
Comment (matiu) [inline src/js/controllers/topup.js]: Maybe this function should be `calculateAmount`...

```
if (useSendMax)  {
  ...
} else 
  return cb(null, amount);
```

and on the result of the function use `txFormatService.parseAmount`

Comment (cmgustavo) [inline src/js/controllers/topup.js]: Good idea. Thanks

**Comment by cmgustavo** [comment]
Updated.

**Pull request merged**